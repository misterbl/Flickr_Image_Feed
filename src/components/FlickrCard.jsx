import React from "react";

export class FlickrCard extends React.Component {
  state = {
    showMore: false
  };

  showMoreorLess = () => {
    this.setState({ showMore: !this.state.showMore });
  };
  render() {
    const {
      image,
      title,
      author,
      author_id,
      link,
      description,
      tags
    } = this.props;
    const divHeight = this.state.showMore ? "" : " mh-500";
    const buttonText = this.state.showMore ? "show less" : "show more";
    return (
      <div>
        <div
          className={`ba pa3 mw5 mr4 mt4 overflow-hidden card-container ${divHeight}`}
        >
          <div className="image-container">
            <img className="image-card" src={image} alt={title} />
          </div>
          <p>
            <a href={link}>{`${title}`}</a> by
            <a
              href={`https://www.flickr.com/photos/${author_id}`}
            >{` ${author}`}</a>
          </p>
          <p
            dangerouslySetInnerHTML={{ __html: `Description ${description}` }}
            className="mt4 no-underline"
          />

          {tags.length > 0 && <p>Tags: {tags}</p>}
        </div>
        <button className="show-more-less-button" onClick={this.showMoreorLess}>
          {buttonText}
        </button>
      </div>
    );
  }
}
export default FlickrCard;
