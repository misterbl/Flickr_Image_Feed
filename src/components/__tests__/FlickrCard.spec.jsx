import * as React from "react";
import { shallow } from "enzyme";
import FlickrCard from "../FlickrCard";

describe("FlickrCard", () => {
  const props = {
    image: "imageTest",
    title: "titleTest",
    author: "authorTest",
    author_id: "123Test",
    link: "linkTest",
    description: "descriptionTest",
    tags: "tagsTest"
  };

  const wrapper = shallow(<FlickrCard {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("changes the state.showmore from false to true on clicking the button", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.state().showMore).toBeTruthy();
  });
  it("changes the state.showmore from true to false on clicking the button", () => {
    wrapper.setState({ showMore: true });
    wrapper.find("button").simulate("click");
    expect(wrapper.state().showMore).toBeFalsy();
  });
});
