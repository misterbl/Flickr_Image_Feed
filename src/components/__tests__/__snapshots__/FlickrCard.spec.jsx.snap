// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlickrCard matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FlickrCard
    author="authorTest"
    author_id="123Test"
    description="descriptionTest"
    image="imageTest"
    link="linkTest"
    tags="tagsTest"
    title="titleTest"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ba pa3 mw5 mr4 mt4 overflow-hidden card-container  mh-500"
        >
          <div
            className="img-block "
          >
            <img
              alt="titleTest"
              src="imageTest"
            />
          </div>
          <p>
            <a
              href="linkTest"
            >
              titleTest
            </a>
             by
            <a
              href="https://www.flickr.com/photos/123Test"
            >
               authorTest
            </a>
          </p>
          <p
            className="mt4 no-underline"
            dangerouslySetInnerHTML={
              Object {
                "__html": "descriptionTest",
              }
            }
          />
          <p>
            Tags: 
            tagsTest
          </p>
        </div>,
        <div
          className="mr4"
        >
          <button
            className="card-container--button w-100"
            onClick={[Function]}
          >
            show more
          </button>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="img-block "
            >
              <img
                alt="titleTest"
                src="imageTest"
              />
            </div>,
            <p>
              <a
                href="linkTest"
              >
                titleTest
              </a>
               by
              <a
                href="https://www.flickr.com/photos/123Test"
              >
                 authorTest
              </a>
            </p>,
            <p
              className="mt4 no-underline"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "descriptionTest",
                }
              }
            />,
            <p>
              Tags: 
              tagsTest
            </p>,
          ],
          "className": "ba pa3 mw5 mr4 mt4 overflow-hidden card-container  mh-500",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="titleTest"
                src="imageTest"
              />,
              "className": "img-block ",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "titleTest",
                "src": "imageTest",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  href="linkTest"
                >
                  titleTest
                </a>,
                " by",
                <a
                  href="https://www.flickr.com/photos/123Test"
                >
                   authorTest
                </a>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "titleTest",
                  "href": "linkTest",
                },
                "ref": null,
                "rendered": "titleTest",
                "type": "a",
              },
              " by",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " authorTest",
                  "href": "https://www.flickr.com/photos/123Test",
                },
                "ref": null,
                "rendered": " authorTest",
                "type": "a",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "mt4 no-underline",
              "dangerouslySetInnerHTML": Object {
                "__html": "descriptionTest",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Tags: ",
                "tagsTest",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Tags: ",
              "tagsTest",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="card-container--button w-100"
            onClick={[Function]}
          >
            show more
          </button>,
          "className": "mr4",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "show more",
            "className": "card-container--button w-100",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "show more",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ba pa3 mw5 mr4 mt4 overflow-hidden card-container  mh-500"
          >
            <div
              className="img-block "
            >
              <img
                alt="titleTest"
                src="imageTest"
              />
            </div>
            <p>
              <a
                href="linkTest"
              >
                titleTest
              </a>
               by
              <a
                href="https://www.flickr.com/photos/123Test"
              >
                 authorTest
              </a>
            </p>
            <p
              className="mt4 no-underline"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "descriptionTest",
                }
              }
            />
            <p>
              Tags: 
              tagsTest
            </p>
          </div>,
          <div
            className="mr4"
          >
            <button
              className="card-container--button w-100"
              onClick={[Function]}
            >
              show more
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="img-block "
              >
                <img
                  alt="titleTest"
                  src="imageTest"
                />
              </div>,
              <p>
                <a
                  href="linkTest"
                >
                  titleTest
                </a>
                 by
                <a
                  href="https://www.flickr.com/photos/123Test"
                >
                   authorTest
                </a>
              </p>,
              <p
                className="mt4 no-underline"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "descriptionTest",
                  }
                }
              />,
              <p>
                Tags: 
                tagsTest
              </p>,
            ],
            "className": "ba pa3 mw5 mr4 mt4 overflow-hidden card-container  mh-500",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="titleTest"
                  src="imageTest"
                />,
                "className": "img-block ",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "titleTest",
                  "src": "imageTest",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    href="linkTest"
                  >
                    titleTest
                  </a>,
                  " by",
                  <a
                    href="https://www.flickr.com/photos/123Test"
                  >
                     authorTest
                  </a>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "titleTest",
                    "href": "linkTest",
                  },
                  "ref": null,
                  "rendered": "titleTest",
                  "type": "a",
                },
                " by",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " authorTest",
                    "href": "https://www.flickr.com/photos/123Test",
                  },
                  "ref": null,
                  "rendered": " authorTest",
                  "type": "a",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "mt4 no-underline",
                "dangerouslySetInnerHTML": Object {
                  "__html": "descriptionTest",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Tags: ",
                  "tagsTest",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Tags: ",
                "tagsTest",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="card-container--button w-100"
              onClick={[Function]}
            >
              show more
            </button>,
            "className": "mr4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "show more",
              "className": "card-container--button w-100",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "show more",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
